src/LanguageServer/Impl/LanguageServer.cs

```diff
@@ -308,7 +308,7 @@ private void OnApplyWorkspaceEdit(object sender, ApplyWorkspaceEditEventArgs e)
         public async Task ClearAnalysisCache(CancellationToken cancellationToken) {
             using (_requestTimer.Time("python/clearAnalysisCache"))
             using (await _prioritizer.ConfigurationPriorityAsync(cancellationToken)) {
-                Debug.Assert(_initialized);
+                // Debug.Assert(_initialized);
```

Comment:
Is this call valid when the server isn't yet initialized? I think some caching stuff gets set up during init.

Reply:
Generally deleting is just FS operation having nothing to do with init. The reason it is commented out is that `LsDeleteAnalysisCache` test that needs to do `ls.Start` but fully initializing the server is now a long envdeavour.

Reply:
Sure, my point was that:

```csharp
_services.AddService(new ModuleDatabase(_services));
```

Is in `initialized`, and `ClearAnalysisCache` fetches this service and uses it without any checks, so there's potential for this call to break in an edge case if the command is sent too early.

Reply:
It is just AF which is not affecting real life case. in RL it won't go through b/c of the prioritizer. In test, it makes it less cumbersome since it takes some acrobatics to have server fully initialized.